applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -633,6 +644,133 @@ void NavierStokesWallCondition<TDim,TNumNodes>::ComputeGaussPointBehrSlipRHSCont
     KRATOS_CATCH("")
 }
 
+
+
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim,TNumNodes>::ComputeGaussPointNavierSlipRHSContribution( VectorType& rRightHandSideVector,
+                                                                                            const ConditionDataStruct& rDataStruct )
+{
+    KRATOS_TRY
+
+    GeometryType& rGeom = this->GetGeometry();
+
+    array_1d<double,TNumNodes> nodal_beta = ZeroVector(TNumNodes);
```

Comment:
The allocation of this array can be avoided as well as the nodes loop below. You can compute such `nodal_beta` inside the loop that computes the RHS contribution.

Reply:
Ok. Done. (also for the LHS)